### ISSUE TITLE
Reverse Proxy

### ISSUE THREAD

**Issue by BeefyAndTheDucks** [SubnauticaNitrox/Nitrox#2147]
### Describe the issue

Is reversed proxy planned for nitrox? Reverse Proxy is another way of network communication without opening ports/using UPnP.

**Comment by Measurity** [comment]
I don't think this would help. The server running the reverse proxy would still need to be port forwarded. If I misunderstood, please point me where I can read up on what you mean.

**Comment by BeefyAndTheDucks** [comment]
How about NAT Punchthrough? It seems that you can connect two clients without port forwarding.

**Comment by Measurity** [comment]
Normally NAT hole punching is done in collaboration with a "rendezvous server". The server is port forwarded and clients connect to it to establish a direct P2P connection afterwards ([see: 3.4 Peers Behind Different NATs](https://bford.info/pub/net/p2pnat/)). More convenient would be to rely on [Steam to do this](https://github.com/ValveSoftware/GameNetworkingSockets/blob/master/README_P2P.md) for us. I'll rename the title so anyone interested in implementing can track this issue.

(I think it would be possible to do UDP NAT punching without a server. But it won't always work depending on routers used. Better to put work on using Steam P2P)

**Comment by BeefyAndTheDucks** [comment]
Is it possible to implement Steam P2P within the mod? If not, I've seen other mods for other games (Nebula for Dyson Sphere Program specifically) use something called ngrok to avoid Port forward and UPnP.

**Comment by Andrew1431** [comment]
It wouldn't be if you're going to continue supporting cross platform play.
